### PR TITLE
refactor: Split RoleManager into DomainManager

### PR DESCRIPTION
**Pull request by Dobmod**
- Maintain consistency with core library
- Add `GetDomainsForUser` API
- Prepare for RBAC with Conditions
- Simplify the `replaceEvalWithMap` function
- Standardize styles (e.g., `Closure`, `count`)
- Improve unit tests

### CODE REVIEW

File: src/Rbac/DefaultRoleManager/RoleManager.php [hunk 1]
```diff
@@ -439,21 +404,38 @@ public function getDomains(string $name): array
...
-     * @return Roles
+     * @param array &$users
+     * @param Closure $fn
      */
-    protected function &loadOrStoreRoles(string $domain, Roles $roles): Roles
+    public function rangeLinks(array &$users, Closure $fn): void
     {
-        if (!isset($this->allDomains[$domain])) {
-            $this->allDomains[$domain] = $roles;
+        foreach ($users as $_ => &$user) {
```
Comment (leeqvip) [inline src/Rbac/DefaultRoleManager/RoleManager.php:425]: Why not just omit it?
```
 foreach ($users as &$user) {
```

File: src/Rbac/DefaultRoleManager/Traits/DomainManager.php [hunk 2]
```diff
@@ -0,0 +1,344 @@
...
+        $this->rmMap = [];
+    }
+
+    /**
+     * Gets the domain from the given arguments.
+     *
+     * @param string ...$domains
+     * @return string
+     */
+    public function getDomain(string ...$domains): string
```
Comment (leeqvip) [inline src/Rbac/DefaultRoleManager/Traits/DomainManager.php:112]: Please use the default value directly
```
public function getDomain(?string $domain = null): string
```

**Review by leeqvip — CHANGES_REQUESTED** [review]

File: src/Rbac/DefaultRoleManager/Role.php [hunk 3]
```diff
@@ -168,8 +189,22 @@ public function toString(): string
...
+    /**
+     * @return string[]
+     */
+    public function getUsers(): array
+    {
+        $names = [];
+        $this->rangeUsers(function ($name, $user) use (&$names) {
+            $names[] = $name;
+        });
+        return $names;
```
[the remark this answers is not in the document]
Comment (Dobmod) [reply src/Rbac/DefaultRoleManager/Role.php:208]: This must use `use` to pass references, and `arrow functions` cannot modify references to external variables. 😊

**Review by leeqvip — APPROVED** [review]

**Pull request merged**